packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -244,46 +244,49 @@ export async function makeWallet({
     return proposal;
   };
 
-  async function updateInboxState(id, offer) {
+  async function updateInboxState(id, offer, doPush = true) {
     // Only sent the uncompiled offer to the client.
-    const {
-      instanceHandleBoardId,
-      installationHandleBoardId,
-      proposalTemplate,
-    } = offer;
+    const { proposalTemplate } = offer;
+    const { instance, installation } = idToOffer.get(id);
+    if (!instance || !installation) {
+      return;
+    }
     // We could get the instanceHandle and installationHandle from the
     // board and store them to prevent having to make this call each
     // time, but if we want the offers to be able to sent to the
     // frontend, we cannot store the instanceHandle and
     // installationHandle in these offer objects because the handles
     // are presences and we don't wish to send presences to the
     // frontend.
```

Comment:
old comment?

Reply:
I've updated it to explain the `action: undefined` etc below.